src/Resources/app/administration/src/module/sw-in-app-purchases/component/sw-in-app-purchase-checkout/index.ts

```diff
@@ -24,11 +28,17 @@ export default Shopware.Component.wrapComponentConfig({
             inAppPurchaseCart: null as IAP.InAppPurchaseCart | null,
             extension: null as IAP.Extension | null,
             tosAccepted: false,
-            errorSnippet: null as string | null
+            errorSnippet: null as string | null,
+            storeExtension: '' as string
         };
     },
 
     created() {
+        // @deprecated tag:v4.0.0 - Will be removed as this.store.extension will be only a string in shopware 6.7
+        const extension = this.store.extension as StoreExtension | string;
+        this.storeExtension = typeof extension === 'object' && !!extension
```

Comment:
if `typeof extension === 'object'`, it will always be trueish

Reply:
This causes the unit test to fail, because for some reason it sees `null` as an `object`

Reply:
I have a better idea: `Shopware.Utils.types.isObject(this.storeExtension)`

Reply:
Really? I HATE JS! Just out of curiosity, does something like this work? `this.storeExtension = extension.name || extension`?